src/main/kotlin/service/user/balance/BalanceManager.kt

```diff
@@ -4,27 +4,19 @@ import model.User
 import model.Users
 import org.jetbrains.exposed.sql.transactions.transaction
 import org.jetbrains.exposed.sql.update
-import utility.RedisFactory
+import utility.RedisConnector
 
 class BalanceManager(val user: User) {
     fun increaseCurrentBalance(increaseAmount: Long = 1) {
-        val redis = RedisFactory.retrieveRedisClient()
-        redis.hincrBy(RedisKeyUserBalances, user.name, increaseAmount)
-        redis.close()
+        RedisConnector().hincrBy(RedisKeyUserBalances, user.name, increaseAmount)
     }
 
     fun decreaseCurrentBalance(decreaseAmount: Long = 1) {
-        val redis = RedisFactory.retrieveRedisClient()
-        redis.hincrBy(RedisKeyUserBalances, user.name, (decreaseAmount * -1))
-        redis.close()
+        RedisConnector().hincrBy(RedisKeyUserBalances, user.name, (decreaseAmount * -1))
     }
 
     fun retrieveCurrentBalance(): Long {
-        val redis = RedisFactory.retrieveRedisClient()
-
-        val currentBalance = redis.hmget(RedisKeyUserBalances, user.name).firstOrNull()?.toLong()
-
-        redis.close()
+        val currentBalance = RedisConnector().hmget(RedisKeyUserBalances, user.name).firstOrNull()?.toLong()
 
         return if (currentBalance != null) {
```

Comment:
Interesting syntax this `return if`. Does it return null instead if the condition is not met?

Reply:
No, it returns the last statement in the `else` block.

Reply:
Oh cool! I guess in kotlin an if statement is an expression, meaning you can put it in lots of places (like with Haskell).